### ISSUE TITLE
Configurable Help source

### ISSUE THREAD

**Issue by thrust26** [stella-emu/stella#948]
Currently all help is provided from the web. It should be possible to configure Stella so that it uses the local help instead.

**Comment by sa666666** [comment]
Maybe instead of using a web address, it can use the local path instead.  The problem here is that it's different for every OS, so that's a complication.

**Comment by thrust26** [comment]
Yes, but I suppose we know the defaults. And we can add a browser window which allows the user to correct the defaults.

**Comment by sa666666** [comment]
We don't know it from within Stella.  For Windows at least, the install path is set by the installer (InnoSetup), which is separate from Stella.  Similarly in Linux, it's determined by the `configure` script.  About the only one we know for certain programatically is macOS, since the files are part of the app bundle.

**Comment by thrust26** [comment]
Would relative paths work?

**Comment by sa666666** [comment]
Only if the user doesn't change where Stella is installed, since that happens at install-time, longer after Stella has been compiled.

I ran into similar issues with `HarmonyCart` software, and where to install the `arm` directory and then have the app auto-magically find it.  In that case, though, I was using Qt, and it helped with this somewhat.  The logic to do this correctly and cross-platform isn't easy.

**Comment by thrust26** [comment]
I mean, wouldn't Stella know where its executable is? Or at least the relative path to the docs?

**Comment by sa666666** [comment]
Not easily in a cross-platform way.  Here's the relevant code from the big 3 OS's:
```
// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
OSystemUNIX::OSystemUNIX()
{
  // Hardcoded for now, since that's what the installer always uses
  // At some point, it would be better to query the OS directly

  myARMPath = "/usr/share/harmonycart/";
}

// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
OSystemMACOS::OSystemMACOS()
{
  QDir dir = QDir(QCoreApplication::applicationDirPath());
  dir.cdUp();
  dir.cdUp();
  dir.cd("arm");
  myARMPath = dir.absolutePath();
}

// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
OSystemWindows::OSystemWindows()
  : OSystem()
{
  QDir dir = QDir(QCoreApplication::applicationDirPath());
  dir.cd("arm");
  myARMPath = dir.absolutePath();
}
```
Windows and macOS are using Qt-specific code, and Linux doesn't even have access to that.  I guess we could look at the Qt source and see how it's done there.  But that still doesn't solve the problem for Linux.

**Comment by thrust26** [comment]
We either find out or let the user navigate to the correct spot. I suppose he should know were the documentation is, no?

At least for Windows we seem to already have the code (`baseinappdir`).

**Comment by thrust26** [comment]
I now remember that Windows is having problems passing parameters when opening a file. It will then simply open the doc, but not navigate to the given context. The user can change this behavior by editing the [registry](https://superuser.com/questions/361816/pass-command-line-arguments-to-windows-open-with), but I suppose that's not feasible here.

How are Linux and MacOS behaving here?

**Comment by sa666666** [comment]
Is this relevant any longer?

**Comment by thrust26** [comment]
It is a nice feature when you are offline, no? Or do you mean the Windows specific problem?
